src/GuppyDataExplorer/ExplorerTable/index.jsx

```diff
@@ -298,7 +339,19 @@ class ExplorerTable extends React.Component {
     return (
       <div className={`explorer-table ${this.props.className}`}>
         {(this.props.isLocked) ? <React.Fragment />
-          : <p className='explorer-table__description'>{explorerTableCaption}</p> }
+          : <div className='explorer-table__description'>
+            <span>
+              {explorerTableCaption}
+            </span>
+            <label className={`explorer-table__hide_empty_column_toggle ${this.props.rawData ? '' : 'ant-switch-disabled'}`}>
+                Show Empty Columns
+              <Switch
+                onChange={this.hideEmptyColumnToggle}
+                disabled={!this.props.rawData}
```

Comment:
should this get disabled when the table gets locked as well?

Reply:
will there be anything in this.props.rawData when the table is locked? i can look into

Reply:
there is a props for that, see https://github.com/uc-cdis/data-portal/blob/2924128c9d9c145fda07e0eb20c02011366d64b5/src/GuppyDataExplorer/ExplorerTable/index.jsx#L358

the table gets locked if the tiered access level is `regular` and current user doesn't have access to all data

Reply:
if you look further up in the code line 341 this code is not shown if its locked
`{(this.props.isLocked) ? <React.Fragment />
          : this code`

Reply:
oops good point!